[docs] add auto-submit info to GH builds guide (#28535)
expo/expo: docs/pages/build/building-from-github.mdx

a build trigger row to disable, edit, or delete the trigger.

<ImageSpotlight
  alt="The options button on the build trigger row"

>   src="/static/images/eas-build/build-from-github/edit-trigger-option.png"

Reviewer: ```suggestion
  src="/static/images/eas-build/build-from-github/edit-trigger-option.png"
  style={{ maxWidth: 320 }}
```

Let's add a width here. Output after this:
![CleanShot 2024-04-30 at 19 08 24@2x](https://github.com/expo/expo/assets/10234615/03a026cd-e438-46c5-ad75-51e74f89fcae)

/>

You can also run a GitHub build with the parameters from the trigger manually. This will not count towards your automatic build trigger record.

> #### Automatic Store Submission with EAS Submit

Reviewer: ```suggestion
#### Automatic app stores submission with EAS Submit
```

> Once your build completes, you can automatically submit your app to the digital stores using EAS Submit. This feature streamlines the process, reducing the manual steps required to publish your app.

Reviewer: ```suggestion
Once your build completes, you can automatically submit your app to the app stores using EAS Submit. This feature streamlines the process, reducing the manual steps required to publish your app.
```

- I think by "digital stores", it means "app stores" here.

To enable automatic submission, you need to configure your build triggers to include submission as part of the build process. Here's how you can set it up:

- Navigate to your project's GitHub settings page on the Expo dashboard.
- Find the build trigger you want to modify, and click the options button.

> - Select "Edit trigger" and in the dialog that appears, check the option "Submit to store after

Reviewer: ```suggestion
- Select **Edit trigger** and in the dialog that appears, check the option **Submit to store after build**.
```

- We use bold names to represent options/settings clicked in Expo dashboard. Let's do the same here for consistency.

>   build".

Reviewer: ```suggestion
```
Added above.

> <ImageSpotlight
>   alt="The EAS Submit form fields on the trigger edit form"
>   src="/static/images/eas-build/build-from-github/eas-submit-form-fields.png"
> />
> 
> - Save your changes.
> 
> <ImageSpotlight
>   alt="An enabled build trigger with automatic submission enabled"
>   src="/static/images/eas-build/build-from-github/auto-submit-trigger-column.png"
> />

Reviewer: ```suggestion
<ImageSpotlight
  alt="The EAS Submit form fields on the trigger edit form"
  src="/static/images/eas-build/build-from-github/eas-submit-form-fields.png"
  style={{ maxWidth: 640 }}
/>

- Save your changes.

<ImageSpotlight
  alt="An enabled build trigger with automatic submission enabled"
  src="/static/images/eas-build/build-from-github/auto-submit-trigger-column.png"
  style={{ maxWidth: 400 }}
/>
```

Let's also add width to the ImageSpotlight components here. Output:

![CleanShot 2024-04-30 at 19 12 47@2x](https://github.com/expo/expo/assets/10234615/4d542b9d-5908-432a-9f5e-bc38f3659e49)

> Once enabled, every time a build is triggered from this configuration, it will automatically be submitted to the store(s) you have configured in your `eas.json` under the `submit` field.

Reviewer: ```suggestion
Once enabled, every time a build is triggered from this configuration, it will automatically be submitted to the app stores you have configured in your **eas.json** under the `submit` field.
```

- We can use app stores as the generic term here.
- eas.json is file name so let's make it bold as per our writing styleguide.

> > **Note:** Ensure that your `eas.json` is properly configured for submission, including specifying
> > the correct store credentials and submission profiles. For more information, see the [EAS Submit
> > documentation](/submit/eas-json/).

Reviewer: ```suggestion
> **Note:** Ensure that your **eas.json** is properly configured for submission, including specifying the correct app store's credentials and submission profile. For more information, see the [EAS Submit](/submit/eas-json/).
```
